src/workerd/api/http.c++

```diff
@@ -989,6 +1127,64 @@ kj::Maybe<kj::String> Request::serializeCfBlobJson(jsg::Lock& js) {
   return cf.serialize(js);
 }
 
+void Request::serialize(
+    jsg::Lock& js, jsg::Serializer& serializer,
+    const jsg::TypeHandler<RequestInitializerDict>& initDictHandler) {
+  serializer.writeLengthDelimited(url);
+
+  // Our strategy is to construct an initializer dict object and serialize that as a JS object.
+  // This makes the deserialization end really simple (just call the constructor), and it also
+  // gives us extensibility: we can add new fields without having to bump the serialization tag.
+  serializer.write(js, jsg::JsValue(initDictHandler.wrap(js, RequestInitializerDict {
+    // GET is the default, so only serialize the method if it's something else.
+    .method = method == kj::HttpMethod::GET ? jsg::Optional<kj::String>() : kj::str(method),
+
+    .headers = headers.addRef(),
+
+    .body = getBody().map([](jsg::Ref<ReadableStream> stream) -> Body::Initializer {
+      // jsg::Ref<ReadableStream> is one of the possible variants of Body::Initializer.
+      return kj::mv(stream);
+    }),
+
+    // "manual" is the default for `redirect`, so only encode if it's not that.
+    .redirect = redirect == Redirect::MANUAL
+        ? kj::str(getRedirect()) : kj::Maybe<kj::String>(kj::none),
+
+    // We have to ignore .fetcher for serialization. We can't simply fail if a fetcher is present
+    // because requests received by the top-level fetch handler actually have .fetcher set to
+    // the hidden "next" binding, which historically could be different from null (although in
+    // practice these days it is always the same). We obviously want to be able to serialize
+    // requests received by the top-level fetch handler so... we have to ignore this. This
+    // property should probably go away in any case.
+
+    .cf = cf.getRef(js),
```

Comment:
We've accidentally broken the lazy parsing of the `cf` property before. It would probably be a good idea to add a test that ensures that the `request.cf` property continues to act as expected after cloning (both on the sending end and the receiving end)

Reply:
807678e5

FWIW if the cf property isn't structured clonable, sending it over RPC will fail here. We know that non-clonable cf properties exist in the wild, but I think this is OK because we're not forcing RPC into any existing scripts.